packages/apps/reputation-oracle/server/src/modules/cron-job/cron-job.service.ts

```diff
@@ -190,35 +302,106 @@ export class CronJobService {
           }
+
+          escrowCompletionTrackingEntity.status =
+            EscrowCompletionTrackingStatus.COMPLETED;
+          await this.escrowCompletionTrackingRepository.updateOne(
+            escrowCompletionTrackingEntity,
+          );
         } catch (err) {
-          this.logger.error(`Error sending webhook: ${err.message}`);
-          await this.webhookService.handleWebhookError(webhookEntity);
+          const errorId = uuidv4();
+          const failedReason = `${ErrorWebhook.PendingProcessingFailed} (Error ID: ${errorId})`;
+          this.logger.error(
+            `Error processing escrow completion tracking. Error ID: ${errorId}, Escrow completion tracking ID: ${escrowCompletionTrackingEntity.id}, Reason: ${failedReason}, Message: ${err.message}`,
+          );
+          await this.webhookService.handleEscrowCompletionTrackingError(
+            escrowCompletionTrackingEntity,
+            failedReason,
+          );
           continue;
         }
-        webhookEntity.status = WebhookStatus.COMPLETED;
+      }
+    } catch (e) {
+      this.logger.error(e);
+    }
+
+    this.logger.log('Pending escrow completion tracking STOP');
+    await this.completeCronJob(cronJob);
+  }
+
+  @Cron('*/2 * * * *')
+  /**
+   * Processes pending outgoing webhooks, sending requests to designated URLs.
+   * Updates each webhook's status upon success, retries or logs errors as necessary.
+   * @returns {Promise<void>} A promise that resolves once all pending outgoing webhooks have been processed.
+   */
+  public async processPendingOutgoingWebhooks(): Promise<void> {
+    const isCronJobRunning = await this.isCronJobRunning(
+      CronJobType.ProcessPendingOutgoingWebhook,
+    );
+
+    if (isCronJobRunning) {
+      return;
+    }
 
-        await this.webhookRepository.updateOne(webhookEntity);
+    this.logger.log('Pending outgoing webhooks START');
+    const cronJob = await this.startCronJob(
+      CronJobType.ProcessPendingOutgoingWebhook,
+    );
+
+    try {
+      const webhookEntities = await this.webhookOutgoingRepository.findByStatus(
+        WebhookOutgoingStatus.PENDING,
+      );
+
+      for (const webhookEntity of webhookEntities) {
+        try {
+          const { url, payload } = webhookEntity;
+
+          await this.webhookService.sendWebhook(url, payload);
+        } catch (err) {
+          const errorId = uuidv4();
+          const failedReason = `${ErrorWebhook.PendingProcessingFailed} (Error ID: ${errorId})`;
+          this.logger.error(
+            `Error processing pending outgoing webhook. Error ID: ${errorId}, Webhook ID: ${webhookEntity.id}, Reason: ${failedReason}, Message: ${err.message}`,
+          );
+          await this.webhookService.handleWebhookOutgoingError(
+            webhookEntity,
+            failedReason,
+          );
+          continue;
+        }
+        webhookEntity.status = WebhookOutgoingStatus.SENT;
+        webhookEntity.retriesCount = 0;
```

Comment:
I guess we can simply remove this line. Can't see a reason why we need to reset this counter.